operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/CustomResourceCache.java

```diff
@@ -3,42 +3,39 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import io.fabric8.kubernetes.client.CustomResource;
-import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.ConcurrentMap;
 import java.util.function.Predicate;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@SuppressWarnings("rawtypes")
 public class CustomResourceCache {
 
   private static final Logger log = LoggerFactory.getLogger(CustomResourceCache.class);
 
-  private ObjectMapper objectMapper = new ObjectMapper();
-  private final Map<String, CustomResource> resources = new ConcurrentHashMap<>();
-  private final Lock lock = new ReentrantLock();
+  private final ObjectMapper objectMapper;
+  private final ConcurrentMap<String, CustomResource> resources = new ConcurrentHashMap<>();
+
+  public CustomResourceCache(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
 
   public void cacheResource(CustomResource resource) {
-    try {
-      lock.lock();
-      resources.put(KubernetesResourceUtils.getUID(resource), resource);
-    } finally {
-      lock.unlock();
-    }
+    resources.put(KubernetesResourceUtils.getUID(resource), resource);
   }
 
   public void cacheResource(CustomResource resource, Predicate<CustomResource> predicate) {
-    try {
-      lock.lock();
-      if (predicate.test(resources.get(KubernetesResourceUtils.getUID(resource)))) {
-        log.trace("Update cache after condition is true: {}", resource);
-        resources.put(resource.getMetadata().getUid(), resource);
-      }
-    } finally {
-      lock.unlock();
-    }
+    resources.compute(
```

Comment:
This change seems unrelated to the PR's focus, especially considering they change the access pattern to the underlying structure so I'd rather not see these changes in this PR. Note that there is already an open issue to replace this cache implementation by a more appropriate one (#356).

Reply:
I've undone the changes. They are definitely unrelated. I just tried to apply the boy scout rule.